sentry-android-ndk/src/main/jni/sentry.c

```diff
@@ -11,8 +11,257 @@ struct transport_options {
 
 struct transport_options g_transport_options;
 
+JNIEXPORT void JNICALL
+Java_io_sentry_android_ndk_NdkScopeObserver_nativeSetTag(JNIEnv *env, jclass cls, jstring key,
+                                                         jstring value) {
+    const char *charKey = (*env)->GetStringUTFChars(env, key, 0);
+    const char *charValue = (*env)->GetStringUTFChars(env, value, 0);
+
+    sentry_set_tag(charKey, charValue);
+}
+
+JNIEXPORT void JNICALL
+Java_io_sentry_android_ndk_NdkScopeObserver_nativeRemoveTag(JNIEnv *env, jclass cls, jstring key) {
+    const char *charKey = (*env)->GetStringUTFChars(env, key, 0);
+
+    sentry_remove_tag(charKey);
+}
+
+JNIEXPORT void JNICALL
+Java_io_sentry_android_ndk_NdkScopeObserver_nativeSetExtra(JNIEnv *env, jclass cls, jstring key,
+                                                           jstring value) {
+    const char *charKey = (*env)->GetStringUTFChars(env, key, 0);
+    const char *charValue = (*env)->GetStringUTFChars(env, value, 0);
+
+    sentry_value_t sentryValue = sentry_value_new_string(charValue);
+    sentry_set_extra(charKey, sentryValue);
+}
+
+JNIEXPORT void JNICALL
+Java_io_sentry_android_ndk_NdkScopeObserver_nativeRemoveExtra(JNIEnv *env, jclass cls,
+                                                              jstring key) {
+    const char *charKey = (*env)->GetStringUTFChars(env, key, 0);
+
+    sentry_remove_extra(charKey);
+}
+
+JNIEXPORT void JNICALL Java_io_sentry_android_ndk_NdkScopeObserver_nativeSetUser(
+        JNIEnv *env,
+        jclass cls,
+        jstring id,
+        jstring email,
+        jstring ipAddress,
+        jstring username) {
+    sentry_value_t user = sentry_value_new_object();
+    if (id) {
+        const char *charId = (*env)->GetStringUTFChars(env, id, 0);
+        sentry_value_set_by_key(user, "id", sentry_value_new_string(charId));
+    }
+    if (email) {
+        const char *charEmail = (*env)->GetStringUTFChars(env, email, 0);
+        sentry_value_set_by_key(
+                user, "email", sentry_value_new_string(charEmail));
+    }
+    if (ipAddress) {
+        const char *charIpAddress = (*env)->GetStringUTFChars(env, ipAddress, 0);
+        sentry_value_set_by_key(
+                user, "ip_address", sentry_value_new_string(charIpAddress));
+    }
+    if (username) {
+        const char *charUsername = (*env)->GetStringUTFChars(env, username, 0);
+        sentry_value_set_by_key(
+                user, "username", sentry_value_new_string(charUsername));
+    }
+    sentry_set_user(user);
+}
+
+JNIEXPORT void JNICALL Java_io_sentry_android_ndk_NdkScopeObserver_nativeRemoveUser(
+        JNIEnv *env,
+        jclass cls) {
+    sentry_remove_user();
+}
+
+//void transform_data_from_java(JNIEnv *env, sentry_value_t crumb, jobject data) {
```

Comment:
parsing the data map as a String seems more performatic than this, so I'd suggest we dont go this way for now.